emily_cron/test/test_deposit_processor.py

```diff
@@ -365,43 +366,191 @@ def tearDownModule():
+
+        # Mock the _enrich_deposits method
+        with patch.object(self.processor, "_enrich_deposits") as mock_enrich:
+            # Return our test deposits when enriching
+            mock_enrich.return_value = [
+                self.expired_locktime,
+                self.rbf_original,
+                self.rbf_replacement,
+            ]
+
+            # Run the update_deposits method
+            self.processor.update_deposits()
+            # Verify the correct API calls were made
+            mock_fetch_deposits.assert_any_call(RequestStatus.PENDING)
+            mock_fetch_deposits.assert_any_call(RequestStatus.ACCEPTED)
+
+            # Verify the enrichment was called with both deposits
+            mock_enrich.assert_called_once()
+
+            # Verify the get_utxo_status was called for the expired_locktime deposit
+            mock_get_utxo_status.assert_called_once()
+
+            # Verify the update was called with the correct updates
+            mock_update_deposits.assert_called_once()
+            updates = mock_update_deposits.call_args[0][0]
+            # We expect 2 updates: one for expired_locktime_tx and one for rbf_original_tx
+            self.assertEqual(len(updates), 2)
+            # Let's check the actual updates to understand what's happening
+            update_txids = [update.bitcoin_txid for update in updates]
+            self.assertIn(
+                "expired_locktime_tx", update_txids, "Should include expired locktime transaction"
+            )
+            self.assertIn(
+                "rbf_original_tx", update_txids, "Should include RBF original transaction"
+            )
+
+            # Count the updates by type
+            expired_locktime_updates = [
+                u for u in updates if u.bitcoin_txid == "expired_locktime_tx"
+            ]
+            rbf_updates = [u for u in updates if u.bitcoin_txid == "rbf_original_tx"]
+
+            self.assertEqual(
+                len(expired_locktime_updates), 1, "Should have 1 expired locktime update"
+            )
+            self.assertEqual(len(rbf_updates), 1, "Should have 1 RBF update")
+
+            # Check the status messages
+            for update in updates:
+                if update.bitcoin_txid == "expired_locktime_tx":
+                    self.assertTrue("Locktime expired" in update.status_message)
+                elif update.bitcoin_txid == "rbf_original_tx":
+                    self.assertTrue("Replaced by confirmed tx" in update.status_message)
+
+    @patch("app.clients.MempoolAPI.get_transaction")
+    @patch("app.clients.MempoolAPI.check_for_rbf")
+    def test_enrich_deposits_with_rbf(self, mock_check_rbf, mock_get_tx):
+        """Test the deposit enrichment process with RBF."""
+        # Create test deposits
+        deposit1 = self._create_mock_deposit(
+            txid="replacement1",
+            confirmed_height=None,
+            lock_time=10,
+            rbf_txids=["replacement1", "replacement2"],
```

Comment:
qq: why do we need `rbf_txids` in `deposit1` and `deposit2`? I would expect them to be populated by `mock_check_rbf`